scripts.js

```diff
@@ -0,0 +1,323 @@
+const edibles = [thingsInHome[0], thingsInHome[1], thingsInHome[5], thingsInGarden[0], thingsInGarden[3]];
+// -- edit above --
+
+console.log(
+    `made out of wood: ${JSON.stringify(madeOutOfWood)}\n edibles: ${JSON.stringify(edibles)}`
+);
+
+// 2
+function getSmallestNumber(numbers) {
+  let smallestNumber = numbers[0];
+
+  for (let i = 1; i < numbers.length; ++i) {
+    const currentNumber = numbers[i];
+
+    if (currentNumber < smallestNumber){
+      smallestNumber = currentNumber;
+    }
+  }
+
+  return smallestNumber;
+}
+
+function getSmallestNumberSort(numbers) {
+  numbers.sort(function(a, b){return a - b});
+  return numbers[0];
+}
+
+getSmallestNumber([2, -5, 10, 1, 4]); // -5
+getSmallestNumber([200, 25, 4, 123, 87]); // 4
+
+console.log(getSmallestNumber([2, -5, 10, 1, 4]));
+console.log(getSmallestNumber([200, 25, 4, 123, 87]));
+
+console.log(getSmallestNumberSort([2, -5, 10, 1, 4]));
+console.log(getSmallestNumberSort([200, 25, 4, 123, 87]));
+
+// 3
+function getSquaredNumbers(numbers) {
+  let squaredNumbers= [];
+
+  for (let i = 0; i < numbers.length; ++i) {
+    squaredNumbers[i] = numbers[i] * numbers[i];
+  }
+
+  return squaredNumbers;
+}
+
+getSquaredNumbers([1, 2, 3, 4, 5]); // [1, 4, 9, 16, 25]
+getSquaredNumbers([6, 7, 8, 9, 10]); // [36, 49, 64, 81, 100]
+
+const numbers = [1, 2, 3];
+const squaredNumbers = getSquaredNumbers(numbers);
+console.log(squaredNumbers); // [1, 4, 9]
+console.log(numbers); // [1, 2, 3]
+console.log(numbers !== squaredNumbers); // true
+
+// 4
+function getReversedString(string) {
+  let characters = string.split('');
+  characters.reverse();
+
+  return characters.join('');
+}
+
+console.log(getReversedString('Hello!')); // '!olleH'
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+//5
+function isPalindrome(string) {
+  string = string.replace(/\s/g, '');
+
+  return string.toLowerCase() === getReversedString(string.toLowerCase());
+}
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')); // true
+console.log(isPalindrome('Hello!')); // false
+
+//6
+function countLetters(string) {
+  let countedLetters= {};
+  let characters = string.replace(/\s/g, '');
+  characters = characters.toLowerCase();
+  characters = characters.split('');
+
+  for (let character of characters) {
+    countedLetters[character] = (countedLetters[character] || 0) + 1;
+  }
+
+  return countedLetters;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+function getYoungestPerson(peopleArray) {
+  let youngestPerson = 100;
+
+  for (let i = 0; i < peopleArray.length; i++) {
+    if (youngestPerson > peopleArray[i].age) {
+      youngestPerson = peopleArray[i].age;
+    }
+  }
+
+  return youngestPerson;
+}
+
+function getOldestPerson(peopleArray) {
```

Comment:
The `getOldestPerson` sounds like it is supposed to return the entire information about a person. In your case, it doesn't

Reply:
You mean the name of the function? That is the given name in homework task. Should I change it?

Reply:
No. The name `getOldestPerson` sounds like it is supposed to return all the information about a person. It means you should return the object containing the oldest person, not just their age.

Reply:
ok, now I get it - fixed